app/signup/invitations/route.js

```diff
@@ -0,0 +1,28 @@
+import Ember from 'ember';
+import DisallowAuthenticated from "diesel/mixins/routes/disallow-authenticated";
+import SignupRouteMixin from "diesel/mixins/routes/signup";
+
+export default Ember.Route.extend(DisallowAuthenticated, SignupRouteMixin, {
+  model(params){
+    let invitationId = params.invitation_id;
+    let verificationCode = params.verification_code;
+
+    return this.store.find('invitation', invitationId).then( (invitation) => {
+      return { invitation, verificationCode };
+    });
+  },
+
+  setupController: function(controller, model){
+    let {invitation, verificationCode} = model;
```

Comment:
I suggest adding an assertion that `model` is not an invitation instance. It seems easy to trip yourself up on.

The model-hook-skipping behavior really sucks here. IMO assertion is fine, the only other idea I can think of is reading params directly off `transition` inside `beforeModel`.